### ISSUE TITLE
Suppress output to stdout in case of 0 issues found

### ISSUE THREAD

**Issue by sdavids** [golangci/golangci-lint#5734]
### Welcome

- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Brew

### Your feature request related to a problem? Please describe

I am not sure if this is a bug, regression, or feature request.

```console
$ golangci-lint --version
golangci-lint has version 2.1.1 built with go1.24.2 from 93dabf9 on 2025-04-12T18:12:29Z
$ golangci-lint run
0 issues.
$ echo $?
0
```

I would like the option to suppress output to stdout in case of no issues found.

### Describe the solution you'd like

```console
$ golangci-lint --version
golangci-lint has version 2.1.1 built with go1.24.2 from 93dabf9 on 2025-04-12T18:12:29Z
$ golangci-lint run
$ echo $?
0
```

Maybe via `-q`/`--quiet` or `-s`/`--silent`.

### Describe alternatives you've considered

`1>/dev/null` and explicit check of `$?`

### Supporter

- [ ] I am a sponsor through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by sdavids** [comment]
I just checked:

```console
$ docker run --rm -it golang:1.24.2
# curl -sOL https://github.com/golangci/golangci-lint/releases/download/v1.64.8/golangci-lint-1.64.8-linux-amd64.deb
# apt install ./golangci-lint-1.64.8-linux-amd64.deb
# printf 'package main\nimport "fmt"\nfunc main(){fmt.Println("Test")}' >src/main.go
# golangci-lint --version
golangci-lint has version 1.64.8 built with go1.24.1 from 8b37f141 on 2025-03-17T20:41:53Z
# golangci-lint run src/main.go
# echo $?
0
```

```console
$ docker run --rm -it golang:1.24.2
# curl -sOL https://github.com/golangci/golangci-lint/releases/download/v2.1.2/golangci-lint-2.1.2-linux-amd64.deb
# apt install ./golangci-lint-2.1.2-linux-amd64.deb
# printf 'package main\nimport "fmt"\nfunc main(){fmt.Println("Test")}' >src/main.go
# golangci-lint --version
golangci-lint has version 2.1.2 built with go1.24.2 from 00a561d9 on 2025-04-15T13:26:47Z
# golangci-lint run src/main.go
0 issues.
# echo $?
0
```

I would say this is either a regression or an undocumented change in behavior.

**Comment by bombsimon** [comment]
You are comparing two major versions, 1.0 and 2.0, so there are a bunch of breaking changes between them and this being one of them. It's mentioned in the [migration guide](https://golangci-lint.run/product/migration-guide/)

> `issues.show-stats`: the existing value is ignored because, in v2, stats are enabled by default.

```sh
› docker run --rm -it golangci/golangci-lint:v1.64.8 /bin/bash
root@7f255dd6f820:/go# printf 'package main\nimport "fmt"\nfunc main(){fmt.Println("Test")}' >src/main.go && cd src
root@7f255dd6f820:/go/src# go mod init x
go: creating new go.mod: module x
root@7f255dd6f820:/go/src# golangci-lint run --show-stats
0 issues.
```

**Comment by deitch** [comment]
I was not as focused on whether or not it is a regression; heaven knows I am responsible for enough software with behavioural changes between major versions that people complained about. 😁 

I was focused on, it is good to have the ability to show stats when there are errors, but to have empty output when there are none. Right now, it is either: show stats whether or not there are errors; or do not show stats whether or not there are errors.

Does that make sense?

**Comment by ldez** [comment]
The "0 issues" is the expected behavior. For now, I don't think adding another option only to hide the "0 issues" is worth it.

**Comment by deitch** [comment]
> expected behaviour

I get that. I am suggesting that another behaviour may be more user-friendly in many cases (and hopefully get other people off your back). Would `--show-stats-errors-only` be bad?

**Comment by ldez** [comment]
I think I answered in the previous comment:

> For now, I don't think adding another option only to hide the "0 issues" is worth it.
